src/Persistence/MassTransit.EntityFrameworkCoreIntegration/EntityFrameworkCoreIntegration/Saga/EntityFrameworkSagaRepositoryContextFactory.cs

```diff
@@ -26,6 +28,62 @@ public class EntityFrameworkSagaRepositoryContextFactory<TSaga> :
             _lockStrategy = lockStrategy;
         }
 
+        public async Task<T> Execute<T>(Func<LoadSagaRepositoryContext<TSaga>, Task<T>> asyncMethod, CancellationToken cancellationToken = default)
```

Comment:
So now there are two methods, cut-n-paste, for load and query. This seems like an unneeded duplication of code for each persistence provider.